### ISSUE TITLE
Rule proposal: `no-concat`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#995]
Main purpose is to stop using `Array#concat()`, but we can't know it's `Array#concat()` or `String#concat()`, and `String#concat()` is not useful anyway.

## Fail

```js
const foo = bar.concat(baz);
```

## Pass

```js
const foo = [...bar, bar];
```

```js
const foo = [...bar, ...baz];
```

```js
const foo = bar + baz; // string concatenation
```

**Comment by sindresorhus** [comment]
Couldn't this just be an addition to the `prefer-spread` rule?

**Comment by fisker** [comment]
Ah... I forgot #737
